library/ratelimiting/shouldRateLimitRequest.ts

```diff
@@ -61,6 +61,14 @@ export function shouldRateLimitRequest(
         maxRequests
       );
 
+    if (context.remoteAddress && context.consumedRateLimitForIP) {
+      agent
+        .getRateLimiter()
+        .decrement(
+          `${endpoint.method}:${endpoint.route}:ip:${context.remoteAddress}`
+        );
```

Comment:
Do we need to reset `consumedRateLimitForIP`? For correctness?

Reply:
Currently it should be fine to do so, but if we allow resetting the user in the future, it could lead to unwanted behavior.